aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -606,13 +606,40 @@ Tensor& floor_divide_(Tensor& self, const Tensor& other) {
 // TODO: Make this structured to undo the perf regression from native:: removal
 // in call here
 Tensor mul(const Tensor& self, const Scalar& other) {
+  // prob ok if self never requires grad
+  if (self.is_zerotensor()) {
+    return self;
+  }
   return at::mul(self, wrapped_scalar_tensor(other)); // redispatch!
 }
 
 Tensor& mul_(Tensor& self, const Scalar& other) {
   return at::mul_out(self, wrapped_scalar_tensor(other), self); // redispatch!
 }
 
+void size_and_device_check(const Tensor& self, const Tensor& other) {
```

Comment:
Is this ever used?

Reply:
yup it's called from `mul_zerotensor` and `add_zerotensor`

Reply:
Not sure how I missed that. ctrl+f failed me.